### PR TITLE
[MX-199] For you horizontal artwork rail

### PR DESCRIPTION
**Pull request by ds300**
https://artsyproduct.atlassian.net/browse/MX-199

horizontal-rail

Also updated the heading style

new-headings

Also:

- Deleted some unused code and assets
- Optimized the initial render of the for-you page
   I noticed that with a cold cache the images above the fold seemed to take forever to load. Upon investigation I found that 350+ images were being fetched as soon as I landed on the for-you page. So I did some FlatList tweaking and now it's more like 50.

#skip_new_tests

### CODE REVIEW

File: src/lib/Components/Home/ArtistRails/ArtistRail.tsx [hunk 1]
```diff
@@ -167,37 +168,42 @@ export class ArtistRail extends Component<Props, State> {
...
-        <CardScrollView>
-          {cards}
-          <View style={{ width: 15 }} />
-        </CardScrollView>
-      )
-    } else {
-      return <Spinner style={{ flex: 1, marginBottom: 20 }} />
-    }
+    return (
+      <FlatList<SuggestedArtist>
```
Comment (ds300) [inline src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: refactored the ArtistRail to use a flat list so we could avoid rendering too many things at once

File: src/lib/Components/Home/ArtistRails/ArtistRail.tsx [hunk 2]
```diff
@@ -167,37 +168,42 @@ export class ArtistRail extends Component<Props, State> {
...
+        ListFooterComponent={() => <Spacer mr={2} />}
+        ItemSeparatorComponent={() => <Spacer mr="15px" />}
+        ListEmptyComponent={() => <Spinner style={{ flex: 1, marginBottom: 20 }} />}
+        horizontal
+        showsHorizontalScrollIndicator={false}
+        scrollsToTop={false}
+        keyExtractor={artist => artist.id}
+        initialNumToRender={2}
+        windowSize={this.state.userHasScrolled ? 4 : 1}
+        onScrollBeginDrag={() => this.setState({ userHasScrolled: true })}
```
Comment (ds300) [inline src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: this is the trick to minimize rendering.

`initialNumToRender` should be just enough to fit on screen, then `windowSize` (measured in screen widths) should be `1` on mount. Then as soon as the user starts to scroll `windowSize` is increased to a normal level.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:

- `src/lib/Scenes/Home/Components/ForYou/Components/__tests__/ArtworkRail-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ds300** [comment]
@ashfurrow just realised this is going to create some significant conflicts with #3041 as they both refactored the artist rail — shall we get that one in first and then I can resolve the conflicts? It would feel so rude to inflict that work on you 😅

**Comment by ashfurrow** [comment]
@ds300 ah yes, that's very kind of you! I'll get that merged right away. We received some feedback from the fairs team about the design, but I think we can merge #3041 now and then apply that feedback afterward.

File: src/lib/Components/SectionTitle.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      )
+    : React.Fragment
+  return (
+    <Wrapper>
+      <Flex mt="3" mb="1" flexDirection="row" alignItems="center">
+        <View style={{ overflow: "hidden", flex: 1 }}>
+          <Sans size="4" ellipsizeMode="tail" numberOfLines={1} data-test-id="title">
+            {title}
+          </Sans>
+          {subtitle && (
```
Comment (ashfurrow) [inline src/lib/Components/SectionTitle.tsx]:
```suggestion
          {Boolean(subtitle) && (
```

**Comment by ds300** [comment]
@ashfurrow conflicts resolved

can't believe how good this looks with the new fairs cards too ❤️ 

image

**Pull request merged**